### PR TITLE
Add setting to hide the tray icon

### LINKED ISSUE #684: Optionally hide tray icon?
Is there a way to optionally hide/not use the tray icon at all? The 'Quit on window close' option does half of what i'd want in this feature suggestion.

Generally, I always have Caprine open, along side several other messenger apps, so the window is not even minimized most of the time and the icon simply adds clutter for me.

How complicated would it be to simply not use an icon and just live from the taskbar entry until closed?

### PR DESCRIPTION
**Pull request by whitecrownclown**
Fixes #684 

@sindresorhus 

I've closed the previous PR, had some issue with those commits, #800 

To summarize:

- Added `Show Tray Icon` menu item, Defaults to `true`
- If `quitOnWindowClose` is `true` or `showTrayIcon` is `false`, don't create tray or `destroy()` already created tray

Thanks!

### CODE REVIEW

**Comment by sindresorhus** [comment]
@CvX Thoughts on this?

File: source/config.ts [hunk 1]
```diff
@@ -19,6 +19,7 @@ const defaults = {
 	},
 	menuBarMode: false,
 	hideDockIcon: false,
+	hideTrayIcon: false,
```
Comment (CvX) [inline source/config.ts]: Can we straighten out both negations in the config? 😃

hideDockIcon -> showDockIcon
hideTrayIcon -> showTrayIcon

Comment (whitecrownclown) [reply source/config.ts]: Done 👍

Comment (CvX) [reply source/config.ts]: Alright, lgtm! 🙂

**Pull request merged**